src/Filters/Graph.php

```diff
@@ -85,34 +85,37 @@ public function calculateForOneGrouping()
             [
                 "title" => "",
                 "values" => $metrics,
-//                "values" => $this->results->pluck($this->getAggregateField()), //now we find the export field
             ]
         ];
         return $this;
     }
 
     public function calculateForTwoGroupings()
     {
-        $dimension = $this->dimension();
+        $aggregatedField = $this->findMetricFieldForOneGrouping();
         $metric = $this->metric();
         $this->labels = $this->results->mapWithKeys(function($row){
             return [(string)$row->{$this->dimensionField->getFilterField()} => $this->dimensionField->getValue($row)];
         })->unique();
         $metrics = $this->results->mapWithKeys(function($row) use($metric){
             return [(string)$row->{$metric} => $this->metricField->getValue($row)];
         });
-        $a = $metrics->map(function($name, $metric) use($dimension) {
-            return ["title" => $name, "values" => $this->labels->map(function($dimensionValue, $dimensionKey) use($dimension, $metric) {
-                $result = $this->results->where($this->metric(), $metric)->first(function($row) use($dimensionValue) {
-                    return $this->dimensionField->getValue($row) == $dimensionValue;
-                });
-                return $result->{$this->getAggregateField()} ?? 0;
-//                return $this->results->where($dimension, $dimensionKey)->where($this->metric(), $metric)->first()->{$this->dimensionField->groupableAggregatedField} ?? 0;
-            })->values()];
+        $a = $metrics->map(function($name, $metric) use ($aggregatedField) {
+            return [
+                "title" => $name,
+                "values" => $this->labels->map(function($dimensionValue) use($metric, $aggregatedField) {
+                    $result = $this->results->where($this->metric(), $metric)->first(function($row) use($dimensionValue) {
+                        return $this->dimensionField->getValue($row) == $dimensionValue;
+                    });
+                    $value = $result->{$this->getAggregateField()} ?? 0;
```

Comment:
return $this->getAggregateField()->getValue($result->{$this->getAggregateField()} ?? 0);